src/main/java/net/sf/jabref/shared/DBMSConnectionProperties.java

```diff
@@ -75,4 +91,45 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    /**
+     *  Gets all required data from {@link SharedDatabasePreferences} and sets them if present.
+     */
+    private void setFromPreferences(SharedDatabasePreferences prefs) {
+        if (prefs.getType().isPresent()) {
+            Optional<DBMSType> dbmsType = DBMSType.fromString(prefs.getType().get());
+            if (dbmsType.isPresent()) {
+                this.type = dbmsType.get();
+            }
+        }
+
+        if (prefs.getHost().isPresent()) {
+            this.host = prefs.getHost().get();
+        }
+
+        if (prefs.getPort().isPresent()) {
+            this.port = Integer.parseInt(prefs.getPort().get());
+        }
+
+        if (prefs.getName().isPresent()) {
+            this.database = prefs.getName().get();
+        }
+
+        if (prefs.getUser().isPresent()) {
+            this.user = prefs.getUser().get();
```

Comment:
Why don't add here ``if(prefs.getPassword.isPresent())``? Then you don't have to check twice if user is present.